src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -1657,9 +1657,7 @@ private static string CreateDisablingMessage(DiagnosticAnalyzer analyzer)
                     diagnosticIds = diagnosticIds.Add(diagnostic.Id);
                 }
             }
-#pragma warning disable CS0618 // ReportIfNonFatalAndCatchUnlessCanceled is obsolete; tracked by https://github.com/dotnet/roslyn/issues/58375
-            catch (Exception e) when (FatalError.ReportIfNonFatalAndCatchUnlessCanceled(e))
-#pragma warning restore CS0618 // ReportIfNonFatalAndCatchUnlessCanceled is obsolete
+            catch (Exception)
```

Comment:
this seems undesirable... pure swallow that gets reported nowhere?

Reply:
@mavasani any suggestions here?

Reply:
Most of the PR seems fine to me, except this bit. Now it won't even get reported.

Reply:
And note that the current behavior here to "report" only would have sent telemetry to us; there's no user indication here at all of what happened.

Reply:
Right, I didn't assume that users would understand what was happening. I meant we wouldn't even be able to understand that there's a problem.

Reply:
So more confusingly, this seems to only be used in CreateAnalyzerExceptionDiagnostic above; it seems in the command line case that disabling message this is producing isn't even shown. It also seems we don't show analyzer exceptions in the IDE _either_, so this might be (accidentally) dead code.

I think the reasonable thing to do here is change the "intentionally empty" to actually return a string saying further errors occurred, but right now I'm unable to prove that string is actually shown anywhere. :-/

Reply:
So further investigation led to https://github.com/dotnet/roslyn/issues/59214. I've updated this to also include the exception text here if we were to get exceptions thrown at this point.

Reply:
@mavasani Any recommendations for where to add a test for this?

Reply:
Chatting with @mavasani privately we'll add a unit test after this goes in, just so we can get this in for our current cutoff.

Reply:
This change did break a test due to the new exception message so I've updated that accordingly.